myia/infer/inferrers.py

```diff
@@ -62,25 +65,83 @@ def getattr_inferrer(node, args, unif):
     return res
 
 
+def getitem_inferrer(node, args, unif):
+    """Inferrer for the getitem function."""
+    obj_node, key_node = args
+    assert key_node.is_constant(int)
+    obj = yield Require(obj_node)
+    key = key_node.value
+    if not (
+        isinstance(obj, data.AbstractStructure)
+        and obj.tracks.interface in (tuple, list)
+    ):
+        raise AssertionError(
+            f"getitem can currently only be used for lists and tuples, got {obj}[{key}]"
+        )
+    return obj.elements[key]
+
+
+def make_tuple_inferrer(node, args, unif):
+    """Inferrer for the make_tuple function."""
+    tuple_types = []
+    for arg_node in args:
+        tuple_types.append((yield Require(arg_node)))
+    return data.AbstractStructure(tuple_types, {"interface": tuple})
+
+
+def make_list_inferrer(node, args, unif):
+    """Inferrer for the make_list function."""
+    tuple_types = []
+    for arg_node in args:
+        tuple_types.append((yield Require(arg_node)))
+    elements = []
+    if tuple_types:
+        from myia.abstract import utils as autils
+
+        base_type = tuple_types[0]
+        for typ in tuple_types[1:]:
+            base_type = autils.merge(base_type, typ, U=unif)[0]
+        elements.append(base_type)
+    return data.AbstractStructure(elements, {"interface": list})
+
+
 X = data.Generic("x")
+Y = data.Generic("Y")
```

Comment:
To be consistent, either capitalize the "x" or lowercase the "Y" (I don't really have a preference).